R/knitr_options.R

```diff
@@ -49,8 +49,11 @@
     }
     
     # re-evaluate code from knitr:::fix_options which is called before the hook
-    if ( !is.null(options$out.width) && is.na(options$out.width) )
-      options$out.width = "100%"
+    #if ( !is.null(options$out.width) && is.na(options$out.width) )
+    #  options$out.width = "100%"
+    
+    if(is.null(options$out.width) && grepl(x = options$fig.env, pattern = "*figure*"))
```

Comment:
The regex will match to any of `smallfigure`, `figure*` (aka wide figure) and `figure`. Assuming it is on purpose like this, I was wondering whether you observed any issues with the latter two figure types?

Reply:
That complete match is intentional.  

I've not actually seen an issue with the wide figures, but knitr is setting a hard width attribute for all figure types.  Since it's not clear to me why the small figures disappear and re-appear based on the width of the page (maybe it's to do with the small screen CSS kicking in?) it doesn't seem unreasonable to assume that something similar might happen to the other figure types on different screen settings.

My understanding of the original code was that it was intended to set the `out.width` to `"100%"` for all figure types, so I've tried to replicate that, and it can always be overridden by the user setting the chunk option explicitly.

Reply:
Makes sense, thanks for the clarification!